File: Scheduler/Samples/Scheduler.MVC3/Controllers/SkinLocaleController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DHTMLX.Scheduler;
using DHTMLX.Scheduler.Data;
using DHTMLX.Scheduler.Controls;
using DHTMLX.Common;
using SchedulerTest.Models;

namespace SchedulerTest.Controllers
{

    /// <summary>
    /// check available skins and localizations
    /// </summary>
    public class SkinLocaleController : Controller
    {
        //
        // GET: /SkinLocale/
        public class LocaleData
        {
            public DHXScheduler scheduler;
            public string locale;
            public string skin;
            public LocaleData(DHXScheduler sched, string loc, string sk)
            {
                scheduler = sched;
                locale = loc;
                skin = sk;
            }
        }
        public ActionResult Index()
        {
            var locale = SchedulerLocalization.Localizations.English;
            var skin = DHXScheduler.Skins.Standart;
            var request_lang = this.Request.QueryString["language"];
            var request_skin = this.Request.QueryString["skin"];

            if (!string.IsNullOrEmpty(request_lang))
            {
                locale = (SchedulerLocalization.Localizations)Enum.Parse(typeof(SchedulerLocalization.Localizations), request_lang);
            }
            if (!string.IsNullOrEmpty(request_skin))
            {
                skin = (DHXScheduler.Skins)Enum.Parse(typeof(DHXScheduler.Skins), request_skin);
            }


            var scheduler = new DHXScheduler(this);
            scheduler.Skin = skin;
            scheduler.Localization.Set(locale);


            scheduler.InitialDate = new DateTime(2011, 11, 24);

            scheduler.XY.scroll_width = 0;
            scheduler.Config.first_hour = 8;
            scheduler.Config.last_hour = 19;
            scheduler.Config.time_step = 30;
            scheduler.Config.multi_day = true;
            scheduler.Config.limit_time_select = true;

            scheduler.Localization.Directory = "locale";


            var rooms = new DHXSchedulerDataContext().Rooms.ToList();

            var unit = new UnitsView("unit1", "room_id");
            unit.AddOptions(rooms);//parse model objects
            scheduler.Views.Add(unit);

            var timeline = new TimelineView("timeline", "room_id");
            timeline.RenderMode = TimelineView.RenderModes.Bar;
            timeline.FitEvents = false;
            timeline.AddOptions(rooms);
            scheduler.Views.Add(timeline);


            scheduler.EnableDataprocessor = true;
            scheduler.LoadData = true;
            scheduler.InitialDate = new DateTime(2011, 9, 19);
            return View(new LocaleData(scheduler, request_lang, request_skin));
        }
        public ContentResult Data()
        {

            var data = new SchedulerAjaxData((new DHXSchedulerDataContext()).Events);
    

            return data;
        }

        public ContentResult Save(int? id, FormCollection actionValues)
        {

            var action = new DataAction(actionValues);

            DHXSchedulerDataContext data = new DHXSchedulerDataContext();
            var changedEvent = DHXEventsHelper.Bind<Event>(actionValues);
            try
            {
                switch (action.Type)
                {
                    case DataActionTypes.Insert:
                        data.Events.InsertOnSubmit(changedEvent);
                        break;
                    case DataActionTypes.Delete:
                        changedEvent = data.Events.SingleOrDefault(ev => ev.id == action.SourceId);
                        data.Events.DeleteOnSubmit(changedEvent);
                        break;
                    default:// "update"                          
                        var eventToUpdate = data.Events.SingleOrDefault(ev => ev.id == action.SourceId);
                        DHXEventsHelper.Update(eventToUpdate, changedEvent, new List<string>() { "id" });
                        break;
                }
                data.SubmitChanges();
                action.TargetId = changedEvent.id;
            }
            catch (Exception a)
            {
                action.Type = DataActionTypes.Error;
            }

            return (new AjaxSaveResponse(action));
        }
    }
}

File: Scheduler/Samples/Scheduler.MVC3.VB/Controllers/SkinLocaleController.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DHTMLX.Scheduler
Imports DHTMLX.Scheduler.Data
Imports DHTMLX.Scheduler.Controls
Imports DHTMLX.Common


Namespace SchedulerTest.Controllers

	''' <summary>
	''' check available skins and localizations
	''' </summary>
	Public Class SkinLocaleController
		Inherits Controller
		'
		' GET: /SkinLocale/
		Public Class LocaleData
			Public scheduler As DHXScheduler
			Public locale As String
			Public skin As String
			Public Sub New(sched As DHXScheduler, loc As String, sk As String)
				scheduler = sched
				locale = loc
				skin = sk
			End Sub
		End Class
		Public Function Index() As ActionResult
            Dim locale = SchedulerLocalization.Localizations.English
            Dim skin = DHXScheduler.Skins.Standart
            Dim request_lang = Me.Request.QueryString("language")
            Dim request_skin = Me.Request.QueryString("skin")

            If (Not String.IsNullOrEmpty(request_lang)) Then
                locale = [Enum].Parse(GetType(SchedulerLocalization.Localizations), request_lang)
            Else
                request_lang = SchedulerLocalization.Localizations.English.ToString()
            End If

            If (Not String.IsNullOrEmpty(request_skin)) Then

                skin = [Enum].Parse(GetType(DHXScheduler.Skins), request_skin)
            Else
                request_skin = DHXScheduler.Skins.Standart.ToString()
            End If
           


            Dim scheduler = New DHXScheduler(Me)
            scheduler.Skin = skin
            scheduler.Localization.Set(locale)


            scheduler.InitialDate = New DateTime(2011, 11, 24)

            scheduler.XY.scroll_width = 0
            scheduler.Config.first_hour = 8
            scheduler.Config.last_hour = 19
            scheduler.Config.time_step = 30
            scheduler.Config.multi_day = True
            scheduler.Config.limit_time_select = True

            scheduler.Localization.Directory = "locale"

            Dim rooms = New DHXSchedulerDataContext().Rooms.ToList()

            Dim unit = New UnitsView("unit1", "room_id")
            unit.AddOptions(rooms)
            'parse model objects
            scheduler.Views.Add(unit)

            Dim timeline = New TimelineView("timeline", "room_id")
            timeline.RenderMode = TimelineView.RenderModes.Bar
            timeline.FitEvents = False
            timeline.AddOptions(rooms)
            scheduler.Views.Add(timeline)


            scheduler.EnableDataprocessor = True
            scheduler.LoadData = True
            scheduler.InitialDate = New DateTime(2011, 9, 19)
            Return View(New LocaleData(scheduler, request_lang, request_skin))
        End Function
		Public Function Data() As ContentResult

			Dim data__1 = New SchedulerAjaxData((New DHXSchedulerDataContext()).Events)


			Return data__1
		End Function

		Public Function Save(id As System.Nullable(Of Integer), actionValues As FormCollection) As ContentResult

			Dim action = New DataAction(actionValues)

			Dim data As New DHXSchedulerDataContext()
			Dim changedEvent = DirectCast(DHXEventsHelper.Bind(GetType([Event]), actionValues), [Event])
			Try
				Select Case action.Type
					Case DataActionTypes.Insert
						data.Events.InsertOnSubmit(changedEvent)
						Exit Select
					Case DataActionTypes.Delete
						changedEvent = data.Events.SingleOrDefault(Function(ev) ev.id = action.SourceId)
						data.Events.DeleteOnSubmit(changedEvent)
						Exit Select
					Case Else
						' "update"                          
						Dim eventToUpdate = data.Events.SingleOrDefault(Function(ev) ev.id = action.SourceId)
						DHXEventsHelper.Update(eventToUpdate, changedEvent, New List(Of String)() From { _
							"id" _
						})
						Exit Select
				End Select
				data.SubmitChanges()
				action.TargetId = changedEvent.id
			Catch a As Exception
				action.Type = DataActionTypes.[Error]
			End Try

			Return (New AjaxSaveResponse(action))
		End Function
	End Class
End Namespace
